lib/file_transfer_agent/file_transfer_agent.js

```diff
@@ -312,6 +318,9 @@ function file_transfer_agent(context) {
     if (largeFileMetas.length > 0) {
       await uploadFilesinSequential(largeFileMetas);
     }
+    if (client.destroy) {
```

Comment:
Is there a reason why you didn't want to use a `try {} finally {}` construct as well? This relates also to line 438.

Reply:
No reason. I missed those two. Added a commit to fix this behavior